src/rgw/rgw_sal_motr.cc

```diff
@@ -2228,14 +2247,12 @@ int MotrObject::remove_mobj_and_index_entry(
         if (rc < 0) {
           ldpp_dout(dpp, 0) <<__func__<< ": ERROR: get_upload_id failed. rc=" << rc << dendl;
         } else {
-          std::string obj_fqdn = bucket_name + "/" + delete_key;
-          std::string obj_part_iname = "motr.rgw.object." + bucket_name + "." +
-                                       this->get_name() + "." + upload_id +
-                                       ".parts";
-          ldpp_dout(dpp, 20) << __func__ << ": object part index=" << obj_part_iname << dendl;
+          std::string obj_fqdn = delete_key + "." + upload_id;
+          std::string iname = "motr.rgw.bucket." + bucket_name + ".multiparts";
+          ldpp_dout(dpp, 20) << __func__ << ": object part index=" << iname << dendl;
           ::Meta *mobj = reinterpret_cast<::Meta*>(&this->meta);
           motr_gc_obj_info gc_obj(upload_id, obj_fqdn, *mobj, std::time(nullptr),
-                                ent.meta.size, true, obj_part_iname);
+                                ent.meta.size, true, iname);
```

Comment:
If I understand it correctly, MotrObject::MotrDeleteOp::create_delete_marker() is called when deleting an object, the code here seems to put the 'bucket' part index into gc queue every time a bucket's object is deleted, shouldn't the index be removed at MotrBucket::remove_bucket()?

Reply:
It is putting part index in GC so that GC can distribute the parts in available GC queues for deletion. Earlier we were deleting this index once all parts are deleted, but with these code changes, we will retain this index until delete bucket op in called.

Reply:
i c, thanks, I mis-understood the code and thought it is to put the index into the gc queue.

Reply:
That's correct. Thank you, @jjxsg, for replying to this.